### DISCUSSION TITLE
Checkboxes for filtering in a view only mechanic

### DISCUSSION THREAD

**Discussion by Fazle-Jahangir** [twilio-labs/paste#2338] [General] [answered]
**Background**

Currently in the exploration phase of adding a feature to Insights Queue Stats where users are able to show or hide columns in a table. Please don't pay too much attention to the copy of the button and user flow in this iteration. 
As shown in the recording, the CTA button opens a side panel which "filters" the selected metrics/columns to be hidden. However, this is not technically filtering as it only affects the display of the table. 

**Question**

Is it appropriate to consider this as a filtering mechanic for only hiding or showing columns? Should the CTA button instead lead to a dashboard component, for example, containing checkboxes that can be selected or deselected to hide or show metrics. I would appreciate your opinion. 

https://user-images.githubusercontent.com/100706574/161304898-0901d02b-cbf8-405f-bb41-da66d77fc3f3.mov

**Comment by andioneto** [comment, marked as answer]
Hey @Fazle-Jahangir 👋 

Thank you for following up with this helpful visual! This is in fact what we suggest. We however do not yet have a side modal component to accommodate this render, which is why we suggest using the popover. Additionally, we suggest inverting the suggested value of the checkbox so that "checked" implies that a column is shown.

For the button copy, our design team suggests the CTA button (that launches the modal) has text similar to "Edit table view" and sit below the filter. Check out the attached image for an example from the video product. 👇 

![customizable_columns](https://user-images.githubusercontent.com/18120906/161646592-30c4accd-fecf-4aaf-9b64-6eb7c01f9d81.png)
